### ISSUE TITLE
Cannot (easily) override standard display renderer

### ISSUE THREAD

**Issue by docwilmot** [backdrop/backdrop-issues#2475]
Because it is hardcoded in `layout_route_handler()`. 
```
  if ($selected_layout) {
    // Render the layout.
    $renderer = layout_create_renderer('standard', $selected_layout);
    ...
    return $renderer->render();
  }
```
To override, you'd need to change the default route handler in `system.core.json` config. But if two modules do that, only one would work.

We could add an optional `$renderer` property to layouts and modify that code block to:
```
  if ($selected_layout) {
    $selected_renderer = isset($selected_layout->renderer) ? $selected_layout->renderer : 'standard';
    $renderer = layout_create_renderer($selected_renderer, $selected_layout);
    ...
    return $renderer->render();
  }
```

This would only work for user-created layouts where that `$renderer` is explicitly set though as there is no obvious way to `_alter` layouts en masse.

**Comment by quicksketch** [comment]
Sounds good to me. Usually when dealing with property that *might* be a string OR a full blown object, I like to have the property include "_name" as the suffix. So instead of `$layout->renderer`, I would recommend `$layout->renderer_name`. Either way we'd want to be sure to define and document the new property in the `Layout` class.

**Comment by jenlampton** [comment]
I'm lost as to what you guys are talking about, though I can tell it has something to do with layouts. Can someone add a use-case here?

**Comment by docwilmot** [comment]
Use case: when building flexible layouts (https://github.com/backdrop-contrib/flexible_layout), I needed to override the display renderer but couldn't. Did a shortcut in a preprocess instead. Still not proud of it. 
In short extending the Standard renderer class is useless without this because core will only ever use Standard.

**Comment by quicksketch** [comment]
PR at https://github.com/backdrop/backdrop/pull/2141 is perfect. Although this is an API expansion, it's very limited in change scope and has no impact on current sites. I didn't get to review this before May 1, but rather than bottleneck this until 1.11, I've merged it in for 1.10, though I'm breaking the feature freeze rule to do so.
